### PR TITLE
extend fix splits tool to report splits with mismatched amounts

### PR DESCRIPTION
**Pull request by matt-fidd**

### CODE REVIEW

File: packages/loot-core/src/server/tools/app.ts [hunk 1]
```diff
@@ -54,9 +57,27 @@
...
+        })
+        .select('*'),
+    )
+  ).data;
+
+  const mismatchedSplits = splitTransactions.filter(t => {
+    const subValue = t.subtransactions.reduce((acc, st) => acc + st.amount, 0);
+
+    return subValue !== t.amount;
+  });
```
[the remark this answers is not in the document]
Comment (matt-fidd) [reply packages/loot-core/src/server/tools/app.ts:74]: The amounts are stored as whole numbers, not decimals

File: packages/desktop-client/src/components/settings/FixSplits.tsx [hunk 2]
```diff
@@ -70,11 +79,7 @@ export function FixSplits() {
       primaryAction={
         <View
           style={{
-            flexDirection: 'row',
-            justifyContent: 'space-between',
-            maxWidth: 500,
-            width: '100%',
-            alignItems: 'center',
+            display: 'block',
```
[the remark this answers is not in the document]
Comment (matt-fidd) [reply packages/desktop-client/src/components/settings/FixSplits.tsx]: Fixed in https://github.com/actualbudget/actual/commit/9b12556bd5c42f11eaf31234c9809330391e0fd1 now

**Review by UnderKoen — COMMENTED** [review]
Shouldn't we create the missing split for them with an note to notify them?

**Comment by matt-fidd** [comment]
> Shouldn't we create the missing split for them with an note to notify them?

I wouldn't want to alter the transactions potentially years back in the data without knowing what the "correct" action is. The end goal here is a modal that pops up and allows the user to correct it there and then, but this is an improvement for the moment.

File: packages/desktop-client/src/components/settings/FixSplits.tsx [hunk 3]
```diff
@@ -15,40 +15,47 @@ import { Setting } from './UI';
...
-        result += '\n';
-      }
-      result += `Fixed ${numDeleted} splits that weren’t properly deleted.`;
+      result.push(`Fixed ${numDeleted} splits that weren’t properly deleted.`);
+    }
+    if (mismatchedSplits.length > 0) {
+      result.push(
+        `Found ${mismatchedSplits.length} split transaction${mismatchedSplits.length > 1 ? 's' : ''} ` +
+          `with mismatched amounts on the below date${mismatchedSplits.length > 1 ? 's' : ''}. ` +
+          `Please fix ${mismatchedSplits.length > 1 ? 'these' : 'this'} manually:\n` +
```
Comment (jfdoming) [inline packages/desktop-client/src/components/settings/FixSplits.tsx:42]:
```suggestion
          `Please review ${mismatchedSplits.length > 1 ? 'them' : 'it'} manually:\n` +
```

**Review by jfdoming — DISMISSED** [review]
LGTM! If you have time, it would be nice to translate the strings in this file

**Comment by matt-fidd** [comment]
> LGTM! If you have time, it would be nice to translate the strings in this file

I've added translations to the dynamic part, and cleaned up the rest of the file into a single <Trans> component. The tenses in the bullet points didn't make much sense in isolation so I've opted to group them for easier understanding. I can split them out if you disagree though.

**Review by jfdoming — DISMISSED** [review]
Thanks! One note is that for things like tenses, it's usually fine to put them in one string as the translation software splits them into "one" and "many" strings if you pass a count

**Comment by matt-fidd** [comment]
> Thanks! One note is that for things like tenses, it's usually fine to put them in one string as the translation software splits them into "one" and "many" strings if you pass a count

Thanks, fixed to use `{{count}}` and only kept the plural as the docs recommend.

**Review by jfdoming — APPROVED** [review]
🙌

**Pull request merged**